mdformat/plugins.py

```diff
@@ -15,3 +20,33 @@ def _load_codeformatters() -> Dict[str, Callable[[str, str], str]]:
 
 
 CODEFORMATTERS: Mapping[str, Callable[[str, str], str]] = _load_codeformatters()
+
+
+class ParsePluginAbstract:
+    """A class for extending the base parser."""
+
+    @staticmethod
+    def update_mdit(mdit: MarkdownIt) -> MarkdownIt:
```

Comment:
I would make this return `None` to make it absolutely clear that this modifies the argument `mdit` in place and does not (and should not) instantiate anything new.

Reply:
yep fair, that change shouldn't anything, although just to note I was actually somewhat copying here the implementation in https://code.visualstudio.com/api/extension-guides/markdown-extension#adding-support-for-new-syntax-with-markdownit-plugins

Reply:
I see. I would still make the change, as I think that's how mutator function interfaces should be (and how e.g. `dict.update()` from stdlib behaves).